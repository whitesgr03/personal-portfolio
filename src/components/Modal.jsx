import { useState } from "react";
const Modal = ({ showProduct, onIfShowProduct }) => {
	const [isModalScroll, setIsModalScroll] = useState(false);
	const [disableModalScroll, setDisableModalScroll] = useState(false);

	const handlePointerOver = () => setDisableModalScroll(true);
	const handlePointerOut = () => setDisableModalScroll(false);

	const handleScroll = e => {
		e.target.scrollTop > e.target.firstElementChild.clientHeight
			? setIsModalScroll(true)
			: setIsModalScroll(false);
	};
	return (
			<ProductModal />
		</div>
	);
};

const ProductModal = () => (
	<div className="productModal">
		<button type="button" className="close" onClick={onCloseModal}></button>
		<div className="describe">
			<h3>Product 1 Product 1 Product 1</h3>
			<div>
				Lorem ipsum dolor sit amet consectetur, adipisicing elit. Modi,
				voluptate.Lorem ipsum dolor sit amet consectetur, adipisicing
				elit. Modi, voluptate.Lorem ipsum dolor sit amet consectetur,
				adipisicing elit. Modi, voluptate.Lorem ipsum dolor sit amet
				consectetur, adipisicing elit. Modi, voluptate.Lorem ipsum dolor
				sit amet consectetur, adipisicing elit. Modi, voluptate.
			</div>
		</div>
		<ul>
			<li>
				<img
					sizes="(max-width: 1920px) 100vw, 1920px"
					srcSet={`https://fakeimg.pl/480x320/195c86/ 480w, https://fakeimg.pl/1004x669/195c86/ 1004w,https://fakeimg.pl/1372x915/195c86/ 1372w,https://fakeimg.pl/1820x1213/195c86/ 1820w`}
					src="https://fakeimg.pl/1920x1280/195c86/"
					alt="product1"
				/>
			</li>
			<li>
				<img
					sizes="(max-width: 1920px) 100vw, 1920px"
					srcSet={`https://fakeimg.pl/480x320/195c86/ 480w, https://fakeimg.pl/1004x669/195c86/ 1004w,https://fakeimg.pl/1372x915/195c86/ 1372w,https://fakeimg.pl/1820x1213/195c86/ 1820w`}
					src="https://fakeimg.pl/1920x1280/195c86/"
					alt="product1"
				/>
			</li>
		</ul>
	</div>
);
export default Modal;
