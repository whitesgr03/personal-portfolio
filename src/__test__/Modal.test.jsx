import { render, screen } from "@testing-library/react";

import Modal, { ProductModal } from "../components/Modal";

describe("Renders Modal Component", () => {
	it("should return modal DOM with show modal", () => {
		const mockShowProduct = true;
		render(<Modal showProduct={mockShowProduct} />);

		const element = screen.getByTestId("modal");

		expect(element).toHaveClass("show");
	});
	it("should return modal DOM without show modal", () => {
		const mockShowProduct = false;
		render(<Modal showProduct={mockShowProduct} />);

		const element = screen.getByTestId("modal");

		expect(element).not.toHaveClass("show");
	});
});

describe("Renders ProductModal Component", () => {
	it("Should return Footer DOM", () => {
		const { container } = render(<ProductModal />);

		expect(container).toMatchSnapshot();
	});
});
