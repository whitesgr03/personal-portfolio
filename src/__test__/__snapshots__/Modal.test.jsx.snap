// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders ProductModal Component Should return Footer DOM 1`] = `
<div>
  <div
    class="productModal"
  >
    <div
      class="describe"
    >
      <h3>
        project1
      </h3>
      <div>
        Lorem ipsum dolor sit amet consectetur, adipisicing elit. Modi, voluptate.Lorem ipsum dolor sit amet consectetur, adipisicing elit. Modi, voluptate.Lorem ipsum dolor sit amet consectetur, adipisicing elit. Modi, voluptate.Lorem ipsum dolor sit amet consectetur, adipisicing elit. Modi, voluptate.Lorem ipsum dolor sit amet consectetur, adipisicing elit. Modi, voluptate.
      </div>
    </div>
    <ul>
      <li>
        <img
          alt="project1"
          sizes="(max-width: 1920px) 100vw, 1920px"
          src="https://fakeimg.pl/1820x1213/195c86/"
          srcset="https://fakeimg.pl/480x320/195c86/ 480w, https://fakeimg.pl/1004x669/195c86/ 1004w, https://fakeimg.pl/1372x915/195c86/ 1372w, https://fakeimg.pl/1820x1213/195c86/ 1820w"
        />
      </li>
      <li>
        <img
          alt="project1"
          sizes="(max-width: 1920px) 100vw, 1920px"
          src="https://fakeimg.pl/1820x1213/195c86/"
          srcset="https://fakeimg.pl/480x320/195c86/ 480w, https://fakeimg.pl/1004x669/195c86/ 1004w, https://fakeimg.pl/1372x915/195c86/ 1372w, https://fakeimg.pl/1820x1213/195c86/ 1820w"
        />
      </li>
      <li>
        <img
          alt="project1"
          sizes="(max-width: 1920px) 100vw, 1920px"
          src="https://fakeimg.pl/1820x1213/195c86/"
          srcset="https://fakeimg.pl/480x320/195c86/ 480w, https://fakeimg.pl/1004x669/195c86/ 1004w, https://fakeimg.pl/1372x915/195c86/ 1372w, https://fakeimg.pl/1820x1213/195c86/ 1820w"
        />
      </li>
    </ul>
  </div>
</div>
`;

exports[`Renders ProductModal Component Should return ProductModal DOM 1`] = `
<div>
  <div
    class="productModal"
  >
    <div
      class="describe"
    >
      <h3>
        project1
      </h3>
      <div>
        Lorem ipsum dolor sit amet consectetur, adipisicing elit. Modi, voluptate.Lorem ipsum dolor sit amet consectetur, adipisicing elit. Modi, voluptate.Lorem ipsum dolor sit amet consectetur, adipisicing elit. Modi, voluptate.Lorem ipsum dolor sit amet consectetur, adipisicing elit. Modi, voluptate.Lorem ipsum dolor sit amet consectetur, adipisicing elit. Modi, voluptate.
      </div>
    </div>
    <ul>
      <li>
        <img
          alt="project1"
          sizes="(max-width: 1920px) 100vw, 1920px"
          src="https://fakeimg.pl/1820x1213/195c86/"
          srcset="https://fakeimg.pl/480x320/195c86/ 480w, https://fakeimg.pl/1004x669/195c86/ 1004w, https://fakeimg.pl/1372x915/195c86/ 1372w, https://fakeimg.pl/1820x1213/195c86/ 1820w"
        />
      </li>
      <li>
        <img
          alt="project1"
          sizes="(max-width: 1920px) 100vw, 1920px"
          src="https://fakeimg.pl/1820x1213/195c86/"
          srcset="https://fakeimg.pl/480x320/195c86/ 480w, https://fakeimg.pl/1004x669/195c86/ 1004w, https://fakeimg.pl/1372x915/195c86/ 1372w, https://fakeimg.pl/1820x1213/195c86/ 1820w"
        />
      </li>
      <li>
        <img
          alt="project1"
          sizes="(max-width: 1920px) 100vw, 1920px"
          src="https://fakeimg.pl/1820x1213/195c86/"
          srcset="https://fakeimg.pl/480x320/195c86/ 480w, https://fakeimg.pl/1004x669/195c86/ 1004w, https://fakeimg.pl/1372x915/195c86/ 1372w, https://fakeimg.pl/1820x1213/195c86/ 1820w"
        />
      </li>
    </ul>
  </div>
</div>
`;
